src/poetry/installation/chooser.py

```diff
@@ -61,7 +61,10 @@ def choose_for(self, package: Package) -> Link:
             links_seen += 1
 
             if link.is_wheel:
-                if not self._no_binary_policy.allows(package.name):
+                if not self._no_binary_policy.allows(package.name) and (
+                    not self._only_binary_policy.has_exact_package(package.name)
+                    or self._no_binary_policy.has_exact_package(package.name)
+                ):
```

Comment:
This check should be redundant, we already did `_no_binary_policy_allows()`.  (And similarly later, on the binary policy)

Reply:
@dimbleby 

`self._no_binary_policy.has_exact_package()` is required.  
`not self._no_binary_policy.allows()` means it includes exact package name or `:all:`. We have to distinguish it with `has_exact_package()`.

I agree that it isn't readable and fixed readability: https://github.com/python-poetry/poetry/pull/10278/commits/82792e646f6a3bc013f517e38c3d8e870015c3d3